OutOfSchool/OutOfSchool.WebApi/Extensions/ExceptionMiddlewareExtension.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Net;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using OutOfSchool.WebApi.Models;
+
+namespace OutOfSchool.WebApi.Extensions
+{
+    public class ExceptionMiddlewareExtension
+    {
+        private readonly RequestDelegate next;
+        private readonly ILogger<ExceptionMiddlewareExtension> logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ExceptionMiddlewareExtensions"/> class.
+        /// </summary>
+        public ExceptionMiddlewareExtension(RequestDelegate next, ILogger<ExceptionMiddlewareExtension> logger)
+        {
+            this.next = next;
+            this.logger = logger;
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public async Task InvokeAsync(HttpContext context)
+        {
+            try
+            {
+                await next(context);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError($"Something went wrong: {ex}");
+                await HandleExceptionAsync(context, ex).ConfigureAwait(false);
+            }
+        }
+
+        private Task HandleExceptionAsync(HttpContext context, Exception exception)
+        {
+            context.Response.ContentType = "application/json";
+            context.Response.StatusCode = 500;
+
+            return context.Response.WriteAsync(JsonSerializer.Serialize(new ErrorDetails()
+            {
+                StatusCode = context.Response.StatusCode,
```

Comment:
do we need to specify StatusCode here? we have specified it for context.Response.StatusCode already

Reply:
Firstly, in the line 46, I manually assigned a 500 status code to the context.Response.StatusCode. Then it needs to be returned it as Json.
We can put only an object type inside the Serialize method (in our case it's ErrorDetails. This class is like a plug. It has no meaning at all, only returns exception details).  And during creating new ErrorDetails, I just assigned a value from line 46 to the Status code. 
In my opinion, it's a good approach. 
If you have suggestion to do it in better way and I missed something - please write.